### PR TITLE
Say chat functionality

### PR DESCRIPTION
**Pull request by HikaruM**
Note, I need the testers to test this bit of code, as I cannot fully test locally. 

radius was set at 92.  This radius, when squared, approximately represents the distance between the two side walls of H3.  (The shorter ones).  Radius may be adjusted as needed.

### CODE REVIEW

File: server/channel/src/ZoneManager.h [hunk 1]
```diff
@@ -135,6 +135,7 @@ class ZoneManager
     void BroadcastPacket(const std::shared_ptr<ChannelClientConnection>& client,
         libcomp::Packet& p, bool includeSelf = true);
 
+    void SendToRange(const std::shared_ptr<ChannelClientConnection>& client, libcomp::Packet& p, bool includeSelf = true);
```
Comment (compomega) [inline server/channel/src/ZoneManager.h]: Space after this an add a comment documenting how to use the function.

File: server/channel/src/ZoneManager.cpp [hunk 2]
```diff
@@ -291,6 +291,32 @@ void ZoneManager::BroadcastPacket(const std::shared_ptr<Zone>& zone, libcomp::Pa
...
 
+void ZoneManager::SendToRange(const std::shared_ptr<ChannelClientConnection>& client, libcomp::Packet& p, bool includeSelf)
+{
+    //get all clients in range
+    auto state = client->GetClientState(); //Get sender's client's state
+    auto cState = state->GetCharacterState(); //Get sender's entity state
+    auto MyX = cState->GetDestinationX();  //Get Sending Char's X
+    auto MyY = cState->GetDestinationY();  //Get Sending Char's Y
+    // auto zoneConnections = GetZoneConnections(client, false); //Get Other zone connections
+    auto r = 92;  // @todo: Define value here  (how big?)
```
Comment (compomega) [inline server/channel/src/ZoneManager.cpp]: Make this a constant in the Constants.h header file.

File: server/channel/src/ZoneManager.cpp [hunk 3]
```diff
@@ -291,6 +291,32 @@ void ZoneManager::BroadcastPacket(const std::shared_ptr<Zone>& zone, libcomp::Pa
...
+
+    std::list<std::shared_ptr<libcomp::TcpConnection>> zConnections;
+    for(auto zConnection : GetZoneConnections(client, includeSelf))
+    {
+        //ZoneX is the x-coord of the other character in the zone
+        auto ZoneX = zConnection->GetClientState()->GetCharacterState()->GetDestinationX();
```
Comment (compomega) [inline server/channel/src/ZoneManager.cpp]: Split zConnection->GetClientState()->GetCharacterState() into a temp variable so it's not called twice.

File: server/channel/src/ZoneManager.cpp [hunk 4]
```diff
@@ -291,6 +291,32 @@ void ZoneManager::BroadcastPacket(const std::shared_ptr<Zone>& zone, libcomp::Pa
...
+        //ZoneY is the y-coord of the other character in question.
+        auto ZoneY = zConnection->GetClientState()->GetCharacterState()->GetDestinationY();
+            //This checks to see if the other character is in range of the sender
+        if(pow(r,2) >= pow((MyX-ZoneX),2)+pow((MyY-ZoneY),2))
```
Comment (compomega) [inline server/channel/src/ZoneManager.cpp]: Not 100% sure if using:
auto x = MyX - ZoneX;
auto y = MyY - ZoneY;
if((r*r) >= (x*x)+(y*y))

Would be better. Might depend on the compiler.

**Comment by HikaruM** [comment]
Closing because travis failed.  Will fix the error and some of the other comments as well.

**Pull request closed without merging**